parts/k8s/addons/kubernetesmasteraddons-kubernetes-dashboard-deployment.yaml

```diff
@@ -1,91 +1,183 @@
+
```

Comment:
nit: yay, we only have one file to maintain! can we rename this to something concise like "kubernetes-dashboard.yml" ?

Reply:
It's not just a deployment, so let's at least drop the `-deployment` bit. But every other addon follows the `kubernetesmasteraddons-component-foo-resource.yaml` convention, so maybe we should drop the prefix in general, but in a separate PR?

Reply:
Not every addon! See those elegant `coredns.yaml`, `ip-masq-agent.yaml`, etc...

Reply:
It's true, just the majority... I am persuaded. 👍 

Maybe I'll make a later PR to shorten the names in general.

Reply:
Yeah, I started down that road and realized the better optimization is to consolidate the version directory-specific specs into single per-addon specs (using go templateing funcs to do various per-version changes). Then we only have to change the filenames in a single directory!